### PR TITLE
Support import from GCS structure

### LINKED ISSUE #3297: Allow import of resources from Preservation Storage without a PreservationObject
PreservationObjects get deleted when the resource does, so we should be able to just point at a directory in GCS and import it.

### PR DESCRIPTION
**Pull request by tpendragon**
Closes #3297 

Confirmed that this can import A/V materials as well as ScannedResources.

### CODE REVIEW

File: app/services/preserver/blind_importer.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        member.id
+      rescue Valkyrie::Persistence::ObjectNotFoundError
+        nil
+      end
+    end.compact!
+    output = change_set_persister.save(change_set: importing_change_set)
+    output
+  end
+
+  def import_file_identifiers
```
Comment (hackmastera) [inline app/services/preserver/blind_importer.rb]: looks like we're importing the actual files as part of this method,  not just getting the ids. rename?

File: app/services/preserver/blind_importer.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+  attr_reader :id, :importing_metadata_adapter, :change_set_persister
+  delegate :storage_adapter, to: :change_set_persister
+  def initialize(id:, importing_metadata_adapter:, change_set_persister:)
+    @id = id
+    @importing_metadata_adapter = importing_metadata_adapter
+    @change_set_persister = change_set_persister
+  end
+
+  def import!
```
Comment (hackmastera) [inline app/services/preserver/blind_importer.rb]: comment that it's recursive might be helpful

File: app/services/preserver/blind_importer/file_metadata_adapter/query_service.rb [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    def initialize(adapter:)
+      @adapter = adapter
+    end
+
+    def find_by(id:)
+      file = storage_adapter.find_by(id: storage_id_from_resource_id(id))
+      json = JSON.parse(file.read)
+      attributes = Valkyrie::Persistence::Shared::JSONValueMapper.new(json).result.symbolize_keys
+      resource_processor.call(resource: Valkyrie::Types::Anything[attributes], adapter: adapter)
+    # Rescue NoMethodError because it's what the GCS shrine adapter is throwing
```
[the remark this answers is not in the document]
Comment (tpendragon) [reply app/services/preserver/blind_importer/file_metadata_adapter/query_service.rb]: https://github.com/pulibrary/figgy/issues/3325

**Comment by tpendragon** [comment]
> I need to look at it a bit more to sort out how the context / parent stuff works. maybe some descriptive comments about that at a class level somewhere would be good.

Yeah. I wish I didn't have to do this and could instead attach some kind of hint to the resource itself that the adapter could use, so that rather than an addition to the adapter it could just implement `find_members`.

File: app/services/preserver/blind_importer/file_metadata_adapter.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      @parents = parents
+      @resource_processor = resource_processor
+    end
+
+    def query_service
+      @query_service ||= QueryService.new(adapter: self)
+    end
+
+    def with_context(parent:, **_args)
+      self.class.new(storage_adapter: storage_adapter, parents: parents + [parent])
```
Comment (hackmastera) [inline app/services/preserver/blind_importer/file_metadata_adapter.rb]: why is parents an array? We have a max of 1 parent.

File: app/services/preserver/blind_importer.rb [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    @source_metadata_adapter = source_metadata_adapter
+    @change_set_persister = change_set_persister
+  end
+
+  # Imports the given ID from the source metadata adapter as well as all its
+  # children.
+  def import!
+    import_binary_files
+    member_ids = source_resource.try(:member_ids) || []
+    member_ids.map! do |member_id|
```
Comment (hackmastera) [inline app/services/preserver/blind_importer.rb]: So after mapping from the figgy ids to the cloud storage ids in the query service, now we're mapping back to figgy ids. 👍

File: app/services/preserver/blind_importer/file_metadata_adapter/query_service.rb [hunk 6]
```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+class Preserver::BlindImporter::FileMetadataAdapter
+  class QueryService
```
Comment (hackmastera) [inline app/services/preserver/blind_importer/file_metadata_adapter/query_service.rb]: A typical valkyrie query service will reference an external converter but b/c of the limited scope here it's all rolled into this class. not just in the form of ConvertLocalStorageIDs but also via `Valkyrie::Types::Anything`

**Comment by hackmastera** [comment]
oh, also related to this review I created #3326

**Pull request merged**